### ISSUE TITLE
falconctl occasionally needs `-f` parameter

### ISSUE THREAD

**Issue by RamblingCookieMonster** [104ru/crowdstrike#12]
Hiyo!

Ran into a few scenarios where a previous config won't be handled by the module - generally, output will be something like:

```
Sensor grouping tags are already set, but -f was not specified
ERROR: failed to process the option --tags
```

This happens for `cid` and `tags`, but you could probably catch the error message and `\w` for the param, and try again with `-y` if that makes sense (among other approaches, if you think it's worth handling this scenario).

I could see this for... cases where you want to change `tags`, or for some reason, remove the `aid` if you need to re-associate with crowdstrike, among others.

Cheers!

**Comment by 104ru** [comment]
It seems like enforcing the module parameters during the registration is a safe way to resolve this. I have committed a new version [1.4.2](https://github.com/104ru/crowdstrike/tree/v1.4.2) to github. Please test and let me know if it works. If it does, I will push it to Forge.

**Comment by nvergottini** [comment]
I ran into the issue as well, but with the CID.
```
CID is set, but -f was not specified
ERROR: failed to process the option --cid
```
In this case, I'm applying a manifest to servers in our DMZ using Bolt. I didn't see this last week, but I'm not sure I applied the manifest a second time after the initial install of Falcon Sensor. I do know the server was just upgraded from Falcon Sensor 6.32.x to 6.33.13005.0 earlier today, so I'm not sure if that is the cause. I have other servers I'm managing using PE Console that are running 6.33.13005.0, and I'm not seeing this issue, but I'm also running v1.4.0 of your module and an older version of stdlib in that environment. I will have to try v1.4.1 and the latest stdlib to see if that changes anything.

I tried v1.4.2 from GitHub in my DMZ environment using Bolt and that does resolve the error, but I'm noticing that it is scheduling a refresh of Service[falcon-sensor] on every run, which would not be good in a PE environment.

I will report back once I've tested 1.4.1 and 1.4.2 in my PE environment.

**Comment by 104ru** [comment]
Thanks @nvergottini. Please let me know how it goes. It is odd that you observe the repeated refresh of the service. Please obtain the facter output for falcon_sensor fact on the affected machine if you confirm the issue.

**Comment by nvergottini** [comment]
Facts appears to be the issue. I must be missing something in the use Puppet with Bolt because I'm not seeing the facts from the modules I'm using within the apply block. Strange that the module classes and resources would be available, but not the facts. All I am seeing is the built-in facts.

**Comment by nvergottini** [comment]
Found the issue. I was running the Bolt plan as my regular user and only escalating to root in the apply block. I tried escalating the apply_prep function to root as well, but that didn't work. I have to use run-as on the entire plan for facts to be gathered properly.

I would say, however, that your fix to run with -f does fix the original issue even if that wasn't the exact issue I was dealing with.
